components/RatingStar.js

```diff
@@ -0,0 +1,41 @@
+import styled from "styled-components";
+import { useState } from "react";
+
+const StarRating = ({ rating, onRate }) => {
+  const [hoverRating, setHoverRating] = useState(0);
+
+  return (
+    <Stars>
+      {[1, 2, 3, 4, 5].map((star) => (
+        <Star
+          key={star}
+          filled={star <= (hoverRating || rating)}
+          onClick={() => onRate(star)}
+          onMouseEnter={() => setHoverRating(star)}
+          onMouseLeave={() => setHoverRating(0)}
+        >
+          ★
+        </Star>
+      ))}
+    </Stars>
+  );
+};
+
+export default StarRating;
+
+const Stars = styled.div`
```

Comment:
If you have a list of things make it a ul or ol and style it which ever way you want